src/utils/ros.lua

```diff
@@ -26,8 +26,23 @@ ros.Time = {}
 
 -- function for legalizing a name for ROS resources
 function ros.Names.sanatize(veh_name)
-    -- replace whitespace with underscores and uppercase with lowercase to meet the ROS naming conventions for TF
-    return veh_name:gsub(" ", "_"):lower()
+    -- in order to meet the ROS naming conventions, we need to sanitize illegal names from FS by using
+    -- 1. "%W" to replace every non-alphanumeric character with an underscore "_"
+    -- 2. "+" modifier to get the longest sequence of non-alphanumeric character, i.e. "___" -> "_" (replace consecutive _ with single _)
+    -- 3. :lower() to replace every uppercase letter with lowercase one
+    veh_name = veh_name:gsub("%W+", "_"):lower()
+
+    -- make sure names do not start or end with underscores
+    -- if it starts with an underscore, remove the first character
+    if veh_name:sub(1,1) == "_" then
+        veh_name = veh_name:sub(2,-1)
+    end
```

Comment:
If I understand [Lua for Python Programmers: strings](http://the4thwiki.com/lua/strings.html) correctly, the `-1` should be unnecessary, right? We just want the string, skipping the first character.

What about using `gsub(..)` here again as an alternative? Something like:


```suggestion
veh_name = veh_name:gsub("^_+", "")
```

Reply:
you are right. it's not necessary.

Is `+` necessary here? Since we've already [replaced all the consecutive `_` with single `_` ](https://github.com/tckarenchiang/FS19_modROS/blob/fix_invalid_topic/src/utils/ros.lua#L33).

Reply:
ok, I removed it.